### PR TITLE
[MNOE-890] Reseller's products on landing page

### PR DESCRIPTION
**Pull request by enizor**

### CODE REVIEW

File: src/app/components/mno-apps/mno-app.coffee [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+        apps = response.apps
+
+        # App to be displayed
+        appId = $stateParams.appId
+        vm.app = _.findWhere(apps, { nid: appId })
+        vm.app ||= _.findWhere(apps, { id:  appId })
+
+        $state.go(parentState) unless vm.app?
+        vm.isLoading = false
+        )
```
Comment (manu-d) [inline src/app/components/mno-apps/mno-app.coffee]: Indentation

File: src/app/components/mno-apps/mno-app.html [hunk 2]
```diff
@@ -104,7 +102,7 @@ <h2 class="media-heading">{{vm.app.name}}</h2>
       <div class="col-md-12">
         <div class="well top-buffer-1" ta-bind="text" ng-model="vm.app.description" ta-readonly="disabled"> </div>
       </div>
-    </div>
+      </div>
```
Comment (manu-d) [inline src/app/components/mno-apps/mno-app.html]: Indentation

File: src/app/components/mno-local-products/mno-local-product-listing.coffee [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      # Initialization
+      #====================================
+      vm.$onInit = ->
+        vm.isPublic = vm.isPublic == "true"
+        vm.productState = if vm.isPublic then "public.local_product" else "home.marketplace.local_product"
+        vm.isLoading = true
+        MnoeMarketplace.getApps().then(
+          (response) ->
+            if vm.isPublic
+              vm.products = _.filter(response.products, (product) -> product.local)
```
Comment (manu-d) [inline src/app/components/mno-local-products/mno-local-product-listing.coffee]: Same as https://github.com/maestrano/mnoe-admin-panel/pull/206#discussion_r163138603

File: src/app/components/mno-apps/mno-app.coffee [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+    vm.app = {}
+    vm.searchWord = ""
+    # The already installed app instance of the app, if any
+    vm.appInstance = null
+    # An already installed app, conflicting with the app because it contains a common subcategory
+    # that is not multi instantiable, if any
+    vm.conflictingApp = null
+    # Enabling pricing
+    vm.isPriceShown = if vm.isPublic
+    then MnoeConfig.isMarketplacePricingEnabled()
```
Comment (ouranos) [inline src/app/components/mno-apps/mno-app.coffee]: indentation. Could also use a ternary operator here.

File: src/app/components/mno-local-products/mno-local-product-listing.coffee [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    }
+    controller: (MnoeMarketplace, MnoeConfig) ->
+      vm = this
+
+      #====================================
+      # Initialization
+      #====================================
+      vm.$onInit = ->
+        vm.isPublic = vm.isPublic == "true"
+        vm.productState = if vm.isPublic then "public.local_product" else "home.marketplace.local_product"
```
Comment (ouranos) [inline src/app/components/mno-local-products/mno-local-product-listing.coffee]: `vm.isPublic ? "public.local_product" : "home.marketplace.local_product"`

File: src/app/views/public/landing/landing.controller.coffee [hunk 6]
```diff
@@ -12,16 +16,29 @@ angular.module 'mnoEnterpriseAngular'
           return _.contains(app.categories, vm.selectedCategory)
 
       vm.carouselImageStyle = (app) ->
+        # Products does not have a picture
+        picture = if app.pictures then app.pictures[0] else app.logo
```
Comment (ouranos) [inline src/app/views/public/landing/landing.controller.coffee]: ternary operator

**Comment by MAhsenArif** [comment]
This can be closed as task is continued on https://github.com/maestrano/mno-enterprise-angular/pull/435

**Comment by ouranos** [comment]
To be continued in #435

**Pull request closed without merging**